src/flows/context/api.tsx

```diff
@@ -40,14 +46,18 @@ export const pooledUpdateAPI = (flow: PatchNotebookParams) => {
 
       clearTimeout(reportDecayTimeout)
       reportDecayTimeout = null
-      updateAPI(stagedFlow)
+      updateAPI(stagedFlow).catch(err =>
+        dispatch(notify(notebookEditFail(err.message)))
+      )
 
       stagedFlow = DEFAULT_API_FLOW
     }, REPORT_MAX_WAIT)
   }
 
   reportDecayTimeout = setTimeout(() => {
-    updateAPI(stagedFlow)
+    updateAPI(stagedFlow).catch(err =>
+      dispatch(notify(notebookEditFail(err.message)))
```

Comment:
Ideally we would not need to pass `dispatch` to this function, but would just throw an error so that the `dispatch` can be handled from the React component. However, I understand why you needed to do that here (since `updateAPI` is an async function subordinated to a synchronous function), so there is not a good way for us to throw the error further up the callstack without a meaningful restructure. I think this is fine for this case.

Reply:
I tested in local dev, and this does not seem to give rise to any rendering issues.